### PR TITLE
Fixes #1939. Adding reduce method to can.List

### LINKED ISSUE #1939: can.List should implement a "reduce" method

### PR DESCRIPTION
**Pull request by mjstahl**
#1939

### CODE REVIEW

File: list/list.js [hunk 1]
```diff
@@ -770,6 +770,34 @@ steal("can/util", "can/map", "can/map/bubble.js","can/map/map_helpers.js",functi
...
+		 *      return p + c;
+		 * }); // -> 10
+		 *
+		 * list.reduce(function(p, c, i, l) {
+		 *      return p + c
+	     * }, 10); // -> 20
+		 * ```
+		 */
+		reduce: function(cb, initial) {
+			return Array.prototype.reduce.apply(this.serialize(), arguments);
```
Comment (justinbmeyer) [inline list/list.js]: This won't work in older browsers.

**Comment by justinbmeyer** [comment]
Maybe move this to can/list/reduce and make it work for older browsers.

File: list/list.js (hunk 1 above)
Comment (mjstahl) [inline list/list.js]: Yeah, I just went back and looked at the MDN docs.. and saw IE 9. _sigh_

**Comment by julia-allyce** [comment]
@mjstahl whats happening with this PR?

**Comment by justinbmeyer** [comment]
This shouldn't be added to 2.3.2.  It would be new functionality. It would need to be scheduled for a 2.4.

**Comment by justinbmeyer** [comment]
This probably would work now in 3.0.  Please make this change in can-define.  Thanks!

**Pull request closed without merging**